File: bestTimeToBuyAndSellStocks2.py
#Time Complexity : O(n) where n is the length of prices
#Space Complexity : O(1)
#Did this code successfully run on Leetcode : Yes


class Solution:
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        max_profit = 0
        for i in range(1, len(prices)):
            if prices[i] > prices[i-1]:
                max_profit += prices[i]-prices[i-1]

        return max_profit

File: bestTimeToBuyAndSellStocks2.java
//Time Complexity : O(n)
//Space Complexity : O(1)
//Did this code successfully run on Leetcode :yes
//Any problem you faced while coding this : no

class Solution {
    public int maxProfit(int[] prices) {
        
        
           if(prices.length == 0){
            
            return 0;
        }
        
         int min = prices[0];
		 int profit = 0;
		 
		 for(int i = 1; i < prices.length; i ++) {
			 
			min = Math.min(min, prices[i]);
			 if(prices[i] > min){
                 
                 profit = profit + prices[i] - min;
                 min = prices[i];
             }
			// int temp = prices[i] - min;
			 
			// profit = Math.max(temp, profit);
		 }
		return profit;
    }
}